src/Workspaces/Core/Portable/Utilities/BKTree.Serialization.cs

```diff
@@ -29,27 +29,33 @@ internal void WriteTo(ObjectWriter writer)
 
         internal static BKTree ReadFrom(ObjectReader reader)
         {
-            var concatenatedLowerCaseWords = new char[reader.ReadInt32()];
-            for (var i = 0; i < concatenatedLowerCaseWords.Length; i++)
-            {
-                concatenatedLowerCaseWords[i] = reader.ReadChar();
-            }
-
-            var nodeCount = reader.ReadInt32();
-            var nodes = ImmutableArray.CreateBuilder<Node>(nodeCount);
-            for (var i = 0; i < nodeCount; i++)
-            {
-                nodes.Add(Node.ReadFrom(reader));
+            try {
+                var concatenatedLowerCaseWords = new char[reader.ReadInt32()];
+                for (var i = 0; i < concatenatedLowerCaseWords.Length; i++)
+                {
+                    concatenatedLowerCaseWords[i] = reader.ReadChar();
+                }
+
+                var nodeCount = reader.ReadInt32();
+                var nodes = ImmutableArray.CreateBuilder<Node>(nodeCount);
+                for (var i = 0; i < nodeCount; i++)
+                {
+                    nodes.Add(Node.ReadFrom(reader));
+                }
+
+                var edgeCount = reader.ReadInt32();
+                var edges = ImmutableArray.CreateBuilder<Edge>(edgeCount);
+                for (var i = 0; i < edgeCount; i++)
+                {
+                    edges.Add(Edge.ReadFrom(reader));
+                }
+
+                return new BKTree(concatenatedLowerCaseWords, nodes.MoveToImmutable(), edges.MoveToImmutable());
             }
-
-            var edgeCount = reader.ReadInt32();
-            var edges = ImmutableArray.CreateBuilder<Edge>(edgeCount);
-            for (var i = 0; i < edgeCount; i++)
+            catch
```

Comment:
We don't in any other serialization code. I think the overall assumption is that we shouldn't assume data we cache on disk stays valid.